### PR TITLE
Hello Agent

### LINKED ISSUE #1321: Release local agent as part of Prefect Core
Rather than a separate repo, the local Agent should ship with Prefect Core, in order to minimize frictions to getting started with Prefect Cloud.

For a user who had already set an auth token, connecting to the Prefect Cloud platform could be as easy as:

```bash
prefect agent start
```

(or something)

### PR DESCRIPTION
**Pull request by joshmeek**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Adds Prefect agents into the core repository. Starting with local, kubernetes, and nomad. Also adds a simple CLI command `prefect agent start` for running agents.

Still to come over the next week or so: comprehensive agent testing, documentation, agent paradigm related issues/requests

Closes #1321

## Why is this PR important?
Was previously in a separate repo but it seems more in line with Prefect core by serving as a tool built on top of it

### CODE REVIEW

File: src/prefect/agent/kubernetes/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+The Prefect Kubernetes agent that turns a cluster into a workflow orchestration system.
+
+If running on GKE you may need to execute: `kubectl create clusterrolebinding default-admin --clusterrole cluster-admin --serviceaccount=default:default`
+
+Quick Start:
+
+- Build Dockerfile and push to registry
+- Update the `image` and `PREFECT__CLOUD__AUTH_TOKEN` env value in the deployment yaml
+- Run `kubectl apply -f deployment.yaml`
```
Comment (cicdw) [inline src/prefect/agent/kubernetes/README.md]: Should we introduce a `prefect agent deploy k8s --image= --token=` CLI command to take care of this (in a future PR)?

File: src/prefect/agent/local/agent.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            self.docker_client.ping()
+        except Exception as exc:
+            self.logger.error(
+                "Issue connecting to the Docker daemon. Make sure it is running."
+            )
+            raise exc
+
+    def deploy_flows(self, flow_runs: list) -> None:
+        """
+        Deploy flow runs on to a k8s cluster as jobs
```
Comment (cicdw) [inline src/prefect/agent/local/agent.py]: should change this

File: src/prefect/cli/agent.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    \b
+        $ prefect agent start kubernetes --token MY_TOKEN
+    """
+    pass
+
+
+@agent.command(hidden=True)
+@click.argument("name", default="local")
+@click.option(
+    "--token", "-t", required=False, help="A Prefect Cloud API token.", hidden=True
```
Comment (cicdw) [inline src/prefect/cli/agent.py]: Looks like this flag doesn't do anything?

File: src/prefect/agent/kubernetes/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# k8s-agent
+
+The Prefect Kubernetes agent that turns a cluster into a workflow orchestration system.
+
+If running on GKE you may need to execute: `kubectl create clusterrolebinding default-admin --clusterrole cluster-admin --serviceaccount=default:default`
```
[the remark this answers is not in the document]
Comment (joshmeek) [reply src/prefect/agent/kubernetes/README.md]: In this or in a separate docs PR that I was planning?

File: src/prefect/agent/kubernetes/README.md (hunk 1 above)
Comment (joshmeek) [reply src/prefect/agent/kubernetes/README.md]: Yeah I have a whole plan for added k8s functionality from the CLI, this being one of them

File: src/prefect/agent/kubernetes/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cicdw) [reply src/prefect/agent/kubernetes/README.md]: In this, just one extra sentence?

File: src/prefect/cli/agent.py (hunk 3 above)
Comment (joshmeek) [reply src/prefect/cli/agent.py]: Yeah it's used to overwrite the value in config for that particular agent run if it is specified

File: src/prefect/agent/kubernetes/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# k8s-agent
+
+The Prefect Kubernetes agent that turns a cluster into a workflow execution platform, orchestrated by Prefect Cloud.
+
+If running on GKE you may need to execute: `kubectl create clusterrolebinding default-admin --clusterrole cluster-admin --serviceaccount=default:default`
+
+The agent needs to be able to read, list, and create both pods and jobs. The resource manager aspect needs the same permissions with the added role of being able to delete jobs and pods. A more specific set of permissions will be added in a later PR.
```
Comment (cicdw) [inline src/prefect/agent/kubernetes/README.md]:
```suggestion
The agent needs to be able to read, list, and create both pods and jobs. The resource manager aspect needs the same permissions with the added role of being able to delete jobs and pods. 
```

File: setup.cfg [hunk 6]
```diff
@@ -51,6 +51,8 @@ omit=
     **__init__.py
     *_version.py
     *_siginfo.py
+    # temporary agent omit
+    src/prefect/agent/*
```
Comment (joshmeek) [inline setup.cfg]:
```suggestion
    src/prefect/agent/*
    agent/*
```

File: setup.cfg [hunk 7]
```diff
@@ -51,6 +51,8 @@ omit=
     **__init__.py
     *_version.py
     *_siginfo.py
+    # temporary agent omit
+    agent/*
```
Comment (cicdw) [inline setup.cfg]:
```suggestion
    */agent/*
```

Comment (cicdw) [reply setup.cfg]: try this @joshmeek

**Pull request merged**